code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -314,8 +314,26 @@
 							M = A
 							break
 				if(M)
-					message = "<span class='danger'>slaps [M] across the face. Ouch!</span>"
+
 					playsound(loc, 'sound/effects/snap.ogg', 50, 1)
+					var/show_ssd
+					var/mob/living/carbon/human/H = M
+					if(istype(H)) show_ssd = H.species.show_ssd

```

Comment:
okay this is better but. Invalid use of `H` below this and code duplication once more.